torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3793,6 +3826,75 @@ std::unique_ptr<Function> CompilationUnit::define(
   return fn;
 }
 
+std::vector<Function*> CompilationUnit::define(
+    const c10::optional<c10::QualifiedName>& prefix,
+    const std::vector<Property>& properties,
+    const std::vector<ResolverPtr>& propResolvers,
+    const std::vector<Def>& definitions,
+    const std::vector<ResolverPtr>& defResolvers,
+    const Self* self,
+    bool shouldMangle) {
+  TORCH_INTERNAL_ASSERT(self);
+  TORCH_INTERNAL_ASSERT(definitions.size() == defResolvers.size());
+  TORCH_INTERNAL_ASSERT(properties.size() == propResolvers.size());
+  std::vector<Function*> functions;
+  std::unordered_map<std::string, Function*> function_table;
+
+  for (size_t i = 0; i < properties.size(); i++) {
+    std::pair<std::unique_ptr<Function>, std::unique_ptr<Function>>
+        property_fns = define(
+            prefix,
+            properties[i],
+            propResolvers[i],
+            self,
+            function_table,
+            shouldMangle);
+
+    auto& getter_fn = property_fns.first;
+    auto& setter_fn = property_fns.second;
+
+    const auto& getter_name = getter_fn->name();
+    function_table[getter_name] = getter_fn.get();
+    functions.push_back(getter_fn.get());
+    register_function(std::move(getter_fn));
+
+    if (setter_fn) {
+      const auto& setter_name = setter_fn->name();
+      function_table[setter_name] = setter_fn.get();
+      functions.push_back(setter_fn.get());
+      register_function(std::move(setter_fn));
+    }
+  }
+
+  for (size_t i = 0; i < definitions.size(); i++) {
```

Comment:
this is a copy-pasta from the other define function, can you unify this and the existing one into one implementation ?

Reply:
Done.